packages/spatial/src/physics/systems/PhysicsPreTransformSystem.test.ts

```diff
@@ -0,0 +1,569 @@
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import {
+  Entity,
+  SystemDefinitions,
+  SystemUUID,
+  UUIDComponent,
+  UndefinedEntity,
+  createEngine,
+  createEntity,
+  destroyEngine,
+  getComponent,
+  hasComponent,
+  removeEntity,
+  setComponent
+} from '@ir-engine/ecs'
+import assert from 'assert'
+import { Quaternion, Vector3 } from 'three'
+import { SceneComponent } from '../../renderer/components/SceneComponents'
+import { EntityTreeComponent, iterateEntityNode } from '../../transform/components/EntityTree'
+import { TransformComponent } from '../../transform/components/TransformComponent'
+import { TransformDirtyUpdateSystem, isDirty } from '../../transform/systems/TransformSystem'
+import { Physics, PhysicsWorld } from '../classes/Physics'
+import { assertVecAllApproxNotEq, assertVecApproxEq } from '../classes/Physics.test'
+import { ColliderComponent } from '../components/ColliderComponent'
+import { RigidBodyComponent } from '../components/RigidBodyComponent'
+import { BodyTypes } from '../types/PhysicsTypes'
+import { PhysicsPreTransformSystem, _PhysicsPreTransformFunctions } from './PhysicsPreTransformSystem'
+
+const _rotation = new Quaternion()
+const _position = new Vector3()
+const _scale = new Vector3()
+
+describe('PhysicsPreTransformFunctions', () => {
+  function assertDirty(entity: Entity, id: number = 0): void {
+    if (!id) {
+      assert.equal(TransformComponent.dirtyTransforms[entity], false)
+      return
+    }
+    assert.equal(TransformComponent.dirtyTransforms[entity], true)
+  }
+
+  function assertNotDirty(entity: Entity, id: number = 0): void {
+    if (!id) {
+      assert.equal(TransformComponent.dirtyTransforms[entity], false)
+      return
+    }
+    if (TransformComponent.dirtyTransforms[entity] === undefined) return
+    assert.equal(TransformComponent.dirtyTransforms[entity], false)
+  }
+
+  describe('setDirty', () => {
```

Comment:
A setDirty test is unnecessary, as the function definition exists only to minimize garbage. We only need to test functionality unique to the module.

Reply:
Fixed with 89c3e14